### PR TITLE
@alanxz External auth

### PR DESCRIPTION
**Pull request by dnj12345**
Percolating external auth changes to SimpleAmqpClient library...

### CODE REVIEW

File: src/SimpleAmqpClient/Channel.h [hunk 1]
```diff
@@ -29,6 +29,7 @@
  * ***** END LICENSE BLOCK *****
  */
 
+#include <amqp.h>
```
Comment (alanxz) [inline src/SimpleAmqpClient/Channel.h]: Don't add rabbitmq-c includes to public SimpleAmqpClient headers (it makes the interface leaky).

File: src/SimpleAmqpClient/Channel.h [hunk 2]
```diff
@@ -89,9 +90,11 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable
                         const std::string &username = "guest",
                         const std::string &password = "guest",
                         const std::string &vhost = "/",
-                        int frame_max = 131072)
+                        int frame_max = 131072,
+                        const amqp_sasl_method_enum sasl_method=AMQP_SASL_METHOD_PLAIN)
```
Comment (alanxz) [inline src/SimpleAmqpClient/Channel.h]: The call for this function when sasl_method is something other than PLAIN now becomes strange. What should username/password be set to when AMQP_SASL_METHOD_EXTERNAL? How about a SASL method that takes a different parameter all together.

It probably needs to be a completely different constructor.

Comment (dnj12345) [inline src/SimpleAmqpClient/Channel.h]: password supplied here is irrelevant but the username has to match the common name in the client certificate. Wanted to keep the api same as much as possible.

Comment (alanxz) [inline src/SimpleAmqpClient/Channel.h]: I'm confused by that: why should the username need to match the CN in the client cert?

Comment (alanxz) [inline src/SimpleAmqpClient/Channel.h]: - Don't change this function signature.
- Add new constructors that allow the user to authenticate using the EXTERNAL mechanism.
- The SimpleAmqpClient headers should be useable without needing the rabbitmq-c headers installed (IOW: don't `#include <amqp.h>` in the public SimpleAmqpClient headers or use things defined in the rabbitmq-c headers in the public SimpleAmqpClient headers.  It is ok to use these in the .cpp and non-public SimpleAmqpClient headers).

Comment (dnj12345) [inline src/SimpleAmqpClient/Channel.h]: the DoLogin which calls amqp_login_with_properties needs username and password.
See the link for ssl authentication mechanism where common name is used as the user name.

https://github.com/rabbitmq/rabbitmq-auth-mechanism-ssl

Comment (alanxz) [inline src/SimpleAmqpClient/Channel.h]: amqp_login_with_properties only needs username and password when using the PLAIN SASL method. Those parameters aren't required for the EXTERNAL method.

**Comment by alanxz** [comment]
This functionality was added in #237

**Pull request closed without merging**